Nos/Views/Modifiers/ReportMenuModifier.swift

```diff
@@ -173,16 +196,37 @@ struct ReportMenuModifier: ViewModifier {
         }
     }
     
-    func mute(author: Author) {
-        Task {
-            do {
-                try await author.mute(viewContext: viewContext)
-            } catch {
-                Log.error(error.localizedDescription)
+    func mute(author: Author) async {
+        do {
+            try await author.mute(viewContext: viewContext)
+        } catch {
+            Log.error(error.localizedDescription)
+        }
+    }
+
+    func unmute(author: Author) async {
+        do {
+            try await author.unmute(viewContext: viewContext)
+        } catch {
+            Log.error(error.localizedDescription)
+        }
+    }
+
+    /// Determines the visibility status for a flagged account and applies the appropriate action.
+    private func determineFlaggedAccoutVisibility() async {
+        if let author = reportedObject.author {
+            guard case .visibility(let visibilityCategory) = selectedVisibilityOption?.category else { return }
+
+            if visibilityCategory == .mute {
+                guard !author.muted else { return }
+                await mute(author: author)
+            } else {
+                guard author.muted else { return }
+                await unmute(author: author)
```

Comment:
I wonder if we want to unmute an author if they're already muted. "Don't Mute" is what the UI says, and to me that's different from "unmute".